### PR TITLE
[Translation] remove duplicate code for loading catalogue.

### PR DESCRIPTION
**Pull request by aitboudad**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Fixed tickets | ~ |
| Tests pass? | yes |
| License | MIT |

### CODE REVIEW

File: src/Symfony/Component/Translation/LoggingTranslator.php [hunk 1]
```diff
@@ -109,16 +109,13 @@ public function __call($method, $args)
      */
     private function log($id, $domain, $locale)
     {
-        if (null === $locale) {
-            $locale = $this->getLocale();
-        }
+        $catalogue = $this->translator->getCatalogue($locale);
```
Comment (stof) [inline src/Symfony/Component/Translation/LoggingTranslator.php]: I would keep the line where it was, to minimize the diff between branches

Comment (aitboudad) [inline src/Symfony/Component/Translation/LoggingTranslator.php]: fixed

File: src/Symfony/Component/Translation/Translator.php [hunk 2]
```diff
@@ -282,6 +262,8 @@ public function getCatalogue($locale = null)
     {
         if (null === $locale) {
             $locale = $this->getLocale();
+        } else {
+            $this->assertValidLocale($locale);
```
Comment (Tobion) [inline src/Symfony/Component/Translation/Translator.php]: getCatalog Interface needs to be updated

Comment (aitboudad) [inline src/Symfony/Component/Translation/Translator.php]: fixed

**Comment by Tobion** [comment]
:+1:

**Comment by fabpot** [comment]
Why is it done in 2.8 and not in 2.7?

**Comment by aitboudad** [comment]
@fabpot well then I split it in 2 PR:
* for 2.6 #14370
* for 2.7 #14371

**Pull request closed without merging**